code/modules/research/protolathe.dm

```diff
@@ -202,3 +202,16 @@
 			if(new_item.matter && new_item.matter.len > 0)
 				for(var/i in new_item.matter)
 					new_item.matter[i] = new_item.matter[i] * mat_efficiency
+
+/obj/machinery/r_n_d/protolathe/proc/eject(var/material, var/amount)

```

Comment:
I'd rather not refactor in a fix. Fix the issue now, refactor later. Like I said, there are plenty of places where they act in the same/similar way - ``process()``, ``dismantle()``, ``attackby()``, etc.

That being said, if you insist that it would be better, I'll change it.